src/js/tabs/gps.js

```diff
@@ -64,12 +64,109 @@ TABS.gps.initialize = function (callback) {
 
-            for (let i = 0; i < FC.GPS_DATA.chn.length; i++) {
-                const row = eSsTable.eq(i);
+            if (FC.GPS_DATA.chn.length <= 16) {
+                for (let i = 0; i < FC.GPS_DATA.chn.length; i++) {
+                    const row = eSsTable.eq(i);
 
-                $('td', row).eq(0).text(FC.GPS_DATA.svid[i]);
-                $('td', row).eq(1).text(FC.GPS_DATA.quality[i]);
-                $('td', row).eq(2).find('progress').val(FC.GPS_DATA.cno[i]);
+                    $('td', row).eq(0).text('-');
+                    $('td', row).eq(1).text(FC.GPS_DATA.svid[i]);
+                    $('td', row).eq(3).text(FC.GPS_DATA.quality[i]);
+                    //$('td', row).eq(3).text(' ');
+                    $('td', row).eq(2).find('progress').val(FC.GPS_DATA.cno[i]);
+                }
+                for (let i = FC.GPS_DATA.chn.length; i < 32; i++) {
+                    const row = eSsTable.eq(i);
+
+                    $('td', row).eq(0).text('-');
+                    $('td', row).eq(1).text('-');
+                    $('td', row).eq(3).text(' ');
+                    $('td', row).eq(2).find('progress').val(0);
+                }
+            } else {
+                for (let i = 0; i < FC.GPS_DATA.chn.length; i++) {
+                    const row = eSsTable.eq(i);
+
+                    switch (FC.GPS_DATA.chn[i]) {
+                        case 0:
+                            $('td', row).eq(0).text('GPS');
+                            break;
+                        case 1:
+                            $('td', row).eq(0).text('SBAS');
+                            break;
+                        case 2:
+                            $('td', row).eq(0).text('Galileo');
+                            break;
+                        case 3:
+                            $('td', row).eq(0).text('BeiDou');
+                            break;
+                        case 4:
+                            $('td', row).eq(0).text('IMES');
+                            break;
+                        case 5:
+                            $('td', row).eq(0).text('QZSS');
+                            break;
+                        case 6:
+                            $('td', row).eq(0).text('GLO');
+                            break;
+                        default:
+                            $('td', row).eq(0).text('-');
+                    }
+
+                    if (FC.GPS_DATA.chn[i] >= 7) {
+                        $('td', row).eq(1).text('-');
+                        $('td', row).eq(3).text(' ');
+                        $('td', row).eq(2).find('progress').val(0);
+                    } else {
+                        $('td', row).eq(1).text(FC.GPS_DATA.svid[i]);
+
+                        let quality = '';
+                        switch (FC.GPS_DATA.quality[i] & 0x7) {
+                            case 0:
+                                quality = quality + 'no signal | ';
```

Comment:
```suggestion
                                quality += 'no signal | ';
```

Reply:
I have reworked those ugly switch cases, so this is no longer an issue :-)